internal/resourcehelper/helper.go

```diff
@@ -0,0 +1,251 @@
+package resourcehelper
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+//go:generate mockgen -source=helper.go -package=resourcehelper -destination=mock_helper_api.go
+
+type Helper interface {
```

Comment:
Why do we need to group those "free" functions under one interface?

If there is a "logical" entity behind that interface then we should have a meaningful name for it, otherwise we don't need an interface.

What is wrong with having "free" helper functions in some helper packages?

Reply:
Reason for this is being able to mock that helper to ease testing of the `Creator`. Another way to achieve that would be to declare functions as a variables, so they could be overwritten in tests (however, this would require some extra overhead on storing old functions and reverting them after test).

Reply:
OK, you convinced me.